File: Haskell/7_max_min.hs
maior :: [Int] -> Int
maior [h] = h
maior (head:tail) | head > maior tail = head
                  | otherwise = maior tail

menor :: [Int] -> Int
menor [h] = h
menor (h:t) | h < menor t = h
            | otherwise   = menor t

max_min :: [Int] -> (Int, Int)
max_min lista = (maior lista, menor lista)

File: List_0/Python/7_max_min.py
#-*- coding=utf-8 -*-
"""Faça um programa que recebe uma lista de inteiros e imprime o maior e o menor
elemento."""

maior = lambda lista: reduce(lambda x, y: x > y and x or y, lista)

menor = lambda lista: reduce(lambda x, y: x < y and x or y, lista)

def show(lista):
	return str(maior(lista)) + ' ' + str(menor(lista))

print show(map(int, raw_input().split()))
